x-pack/metricbeat/module/meraki/network_health_channel_utilization/network_health_channel_utilization.go

```diff
@@ -0,0 +1,199 @@
+
+		networkHealthUtilizations, err := getNetworkHealthChannelUtilization(m.client, orgNetworks)
+		if err != nil {
+			return err
+		}
+
+		reportNetworkHealthChannelUtilization(reporter, org, devices, networkHealthUtilizations)
+
+	}
+
+	return nil
+}
+
+func getNetworkHealthChannelUtilization(client *meraki_api.Client, networks *meraki_api.ResponseOrganizationsGetOrganizationNetworks) ([]*meraki_api.ResponseNetworksGetNetworkNetworkHealthChannelUtilization, error) {
+
+	var networkHealthUtilizations []*meraki_api.ResponseNetworksGetNetworkNetworkHealthChannelUtilization
+
+	for _, network := range *networks {
+
+		for _, product_type := range network.ProductTypes {
+
+			if strings.Compare(product_type, "wireless") == 0 {
+
+				networkHealthUtilization, _, err := client.Networks.GetNetworkNetworkHealthChannelUtilization(network.ID, &meraki_api.GetNetworkNetworkHealthChannelUtilizationQueryParams{})
+				if err != nil {
+					//return nil, fmt.Errorf("Networks.GetNetworkNetworkHealthChannelUtilization failed; [%d] %s. %w", res.StatusCode(), res.Body(), err)
+					//log.Printf("\n#ERROR Networks.GetNetworkNetworkHealthChannelUtilization failed; [%d] %s.", res.StatusCode(), res.Body())
+					//fmt.Printf("\n#ERROR Networks.GetNetworkNetworkHealthChannelUtilization failed; [%d] %s.\n###################", res.StatusCode(), res.Body())
+				} else {
+					//fmt.Printf("\n#DEBUG Networks.GetNetworkNetworkHealthChannelUtilization; [%d] %s. \n###################", res.StatusCode(), res.Body())
+					networkHealthUtilizations = append(networkHealthUtilizations, networkHealthUtilization)
+				}
+
+			}
+		}
+	}
+
+	return networkHealthUtilizations, nil
+}
+
+func reportNetworkHealthChannelUtilization(reporter mb.ReporterV2, organizationID string, devices map[meraki.Serial]*meraki.Device, networkHealthUtilizations []*meraki_api.ResponseNetworksGetNetworkNetworkHealthChannelUtilization) {
+
+	metrics := []mapstr.M{}
+	//Note: API does not specifiy if wireless devices only, so iterating through all network devices. API is to ambiguous
+	//for _, device := range devices {
+
+	//fmt.Printf("\n#DEBUG device Info: \nSerial: %s \nName: %s \nMode: %s \nNetworkID: %s \nProducttype: %s", device.Serial, device.Name, device.Model, device.NetworkID, device.ProductType)
+
+	//fmt.Printf("\n#DEBUG reportNetworkHealthCHannelUtilization")
+
+	for _, networkHealthUtil := range networkHealthUtilizations {
+		//fmt.Printf("\n#DEBUG reportNetworkHealthCHannelUtilization For LOOP 1")
+
+		for _, network := range *networkHealthUtil {
+
+			//fmt.Printf("\n#DEBUG reportNetworkHealthCHannelUtilization For LOOP 2")
+
+			metric := mapstr.M{
+				"network.health.channel.radio.serial": network.Serial,
+				"network.health.channel.radio.model":  network.Model,
+				"network.health.channel.radio.tags":   network.Tags,
+			}
+
+			for _, wifi0 := range *network.Wifi0 {
```

Comment:
Great catch, I can put an iterator into the looping structure, using a sprintf ... you think that is a good way to handle it? https://developer.cisco.com/meraki/api-v1/get-network-network-health-channel-utilization/

Reply:
resolved with iterator, not sure if great practice, I can probably also generate multiple unique metric events, I debate whether sending numerous mb events or iterate ... in the case of ports there was so many dang ports, I just decided to iterate, i think in this case it might only be a few.